src/duckling/entitysystem/entity-system.reducer.ts

```diff
@@ -30,6 +30,35 @@ function _isUpdateEntityAction(action : Action): action is EntityUpdateAction  {
     return action.type === ACTION_UPDATE_ENTITY;
 }
 
+const ACTION_UPDATE_ENTITIES = "EntitySystem.UpdateEntities";
+
+/**
+ *  Action used to describe updates to multiple entities. If no entity with the
+ *  given key exists, then a new entity will be created.
+ */
+export interface EntitiesUpdateAction extends Action {
+    entities : Entity[];
+    entityKeys : EntityKey[];
+}
+
+/**
+ * Factory function for a EntitiesUpdateAction
+ * @param  entities    New value for the entities.
+ * @param  entityKeys The keys for the entities
+ * @return A new EntitiesUpdateAction.
+ */
+export function updateEntitiesAction(entities : Entity[], entityKeys : EntityKey[]) : EntitiesUpdateAction {
+    return {
+        type : ACTION_UPDATE_ENTITIES,
+        entities,
```

Comment:
Fixed. Now using Map<EntityKey, Entity>
I considered making a type alias for this in entity.ts, called PartialEntitySystem, but it just seemed unnecessary